File: bookstore-desktop/Bookstore.Desktop/Dtos/CreatePosOrderRequest.cs
namespace Bookstore.Desktop.Dtos;

public class CreatePosOrderRequest
{
    public string? CustomerName { get; init; }
    public string? CustomerPhone { get; init; }
    public string PaymentMethod { get; init; } = "";
    public string? CouponCode { get; init; }
    public IReadOnlyList<CreatePosOrderItemRequest> Items { get; init; } = Array.Empty<CreatePosOrderItemRequest>();
}

File: bookstore-backend/src/main/java/com/bookstore/bookstore/presentation/request/CreatePosOrderRequest.java
package com.bookstore.bookstore.presentation.request;

import com.bookstore.bookstore.domain.enums.PaymentMethod;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import java.util.List;

public record CreatePosOrderRequest(
        String customerName,
        String customerPhone,

        @NotNull(message = "paymentMethod không được null")
        PaymentMethod paymentMethod,

        String couponCode,

        @NotEmpty(message = "items không được để trống")
        List<@Valid CreatePosOrderItemRequest> items
) {
}

